content/en/docs/tasks/traffic-management/ingress/secure-ingress/index.md

```diff
@@ -423,12 +423,37 @@ $ kubectl create -n istio-system secret generic httpbin-credential --from-file=t
             `"""`
     {{< /text >}}
 
+## More info
+
+### Key formats
+
 Istio supports reading a few different Secret formats, to support integration with various tools such as [cert-manager](/docs/ops/integrations/certmanager/):
 
 * A TLS Secret with keys `tls.key` and `tls.crt`, as described above. For mutual TLS, a `ca.crt` key can be used.
 * A generic Secret with keys `key` and `cert`. For mutual TLS, a `cacert` key can be used.
 * A generic Secret with keys `key` and `cert`. For mutual TLS, a separate generic Secret named `<secret>-cacert`, with a `cacert` key. For example, `httpbin-credential` has `key` and `cert`, and `httpbin-credential-cacert` has `cacert`.
 
+## SNI Routing
+
+A HTTPS `Gateway` configuring the `hosts` field will perform an [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) match on incoming requests.
+For example, the following configuration would only allow requests that match `*.example.com` in the SNI:
+
+{{< text yaml >}}
+servers:
+- port:
+    number: 443
+    name: https
+    protocol: HTTPS
+  hosts:
+  - "*.example.com"
+{{< /text >}}
+
+This may cause certain requests to fail.
+
+For example, if you do not have DNS set up an are instead directly setting the host header, such as `curl 1.2.3.4 -H "Host: app.example.com"`, no SNI will be set, causing the request to fail. Instead, you can set up DNS or use the `--resolve` flag of `curl`.
+
+Another common issue is load balancers in front of Istio. Most cloud load balancers will not forward the SNI, so if you are terminating TLS in your cloud load balancer you may need to either configure it to instead passthrough the TLS connection, or changes `hosts` to match `*`, which will disable the SNI matching.
```

Comment:
This should probably also (or instead) be in the common problems doc: https://preliminary.istio.io/latest/docs/ops/common-problems/network-issues/

Reply:
I think it should be "also". Its important to be here as well I think

Reply:
Mentioning it in this doc makes sense, but duplicating the same information in both places seems kind of ugly. How about instead of adding this section here, just add a warning around line 95, something like this:
```
{{< warning >}}
An HTTPS `Gateway` with a `hosts` field value other than `*` will perform
[SNI](https://en.wikipedia.org/wiki/Server_Name_Indication) matching before forwarding a request,
which may cause some requests to fail. See [configuring SNI routing](link to common problem) for details.
{{< /warning >}}
```

Reply:
we should also mention the symptom. Health check passes but real traffic gets a 404.

Reply:
Why does the health check succeed? is it just doing connection establishment?

The symptom would be connection closed btw not 404 - we never complete tls handshake. I will add it

Reply:
or I guess the health check is port 15021, nevermind